internal/provider/resource_gitlab_application_settings.go

```diff
@@ -0,0 +1,78 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+const applicationSettingsID = "gitlab"
+
+var _ = registerResource("gitlab_application_settings", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`" + `gitlab_application_settings` + "`" + ` resource allows to manage the GitLabLab application settings.
+
+~> This is an **experimental resource**. By nature it doesn't properly fit into how Terraform resources are meant to work.
+   Feel free to join the [discussion](https://github.com/gitlabhq/terraform-provider-gitlab/issues/957) if you have any
+   ideas or questions regarding this resource.
+
+~> All ` + "`" + `gitlab_application_settings` + "`" + ` use the same ID ` + "`" + `gitlab` + "`" + `.
+
+!> This resource does not implement any destroy logic, it's a no-op at this point.
+   It's also not possible to revert to the previous settings.
+
+-> Requires at administrative privileges on GitLab.
```

Comment:
@timofurrer  - Seems obvious, but should we note that this doesn't work on the SaaS offering as a result just to be explicit? Something like this:

```suggestion
-> Requires at administrative privileges on a self-hosted GitLab instance.
```

Reply:
Nice one 🎉